edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Amqp/EdgeTlsTransport.cs

```diff
@@ -44,7 +44,7 @@ protected override X509Principal CreateX509Principal(X509Certificate2 certificat
         protected override bool ValidateRemoteCertificate(object sender, X509Certificate certificate, X509Chain chain, SslPolicyErrors sslPolicyErrors)
         {
             // copy of the chain elements since they are destroyed after this method completes
-            this.remoteCertificateChain = chain == null ? new List<X509Certificate2>() : chain.ChainElements.Cast<X509ChainElement>().Select(element => element.Certificate).ToList();
+            this.remoteCertificateChain = chain == null ? new List<X509Certificate2>() : chain.ChainElements.Cast<X509ChainElement>().Select(element => new X509Certificate2(element.Certificate)).ToList();
```

Comment:
Please explain why this diff "new X509Certificate2()".

Reply:
Our existing code in master saves a reference to a connecting client's cert chain without deep copying it. This was fine in dotnet 2 but won't work in dotnet 3. We can fix this by deep copying the certs in amqp, mqtt, and https scenarios.